MainWindow.cpp

```diff
@@ -49,7 +53,28 @@ MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), ui(new Ui::MainWi
 
 MainWindow::~MainWindow() { delete ui; }
 
-void MainWindow::closeEvent(QCloseEvent *event) { event->accept(); }
+void MainWindow::readSettings() {
+  QSettings settings;
+
+  settings.beginGroup("MainWindow");

```

Comment:
I only know what the hell this is doing because we just talked about it. Can you add a comment saying, eg,`// Read editor state / window geometry`? Similar for below. Nothing elaborate. The name of this function suggests you're reading user preferences, not just trivia about the way the IDE was last time it was closed.

Reply:
I've added a comment clarifying it for read and write.